### PR TITLE
Add script to fetch missing deployment logs (adds missing logs for worldchain)

### PR DESCRIPTION
**Pull request by ezynda3**
# Which Jira task belongs to this PR?

# Why did I implement it this way?

# Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] This pull request is as small as possible and only tackles one problem
- [ ] I have added tests that cover the functionality / test the bug
- [ ] I have updated any required documentation

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

### CODE REVIEW

**Review by 0xDEnYO — APPROVED** [review]

File: deployments/_deployments_log_file.json [hunk 1]
```diff
@@ -690,6 +690,20 @@
           }
         ]
       }
+    },
+    "worldchain": {
+      "production": {
+        "2.0.0": [
+          {
+            "ADDRESS": "0x57FdfF2e36De6c8a8Cde297B150Ae291132Eae8d",
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply deployments/_deployments_log_file.json:698]: Not a problem. We can have the address on different chains.

File: script/utils/fetch-missing-deployments.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+          masterLog[contractName][network].production = {}
+        }
+        if (!masterLog[contractName][network].production[version]) {
+          masterLog[contractName][network].production[version] = [
+            {
+              ADDRESS: contractAddress,
+              OPTIMIZER_RUNS: data.result[0].Runs || 0,
+              TIMESTAMP: new Date().toISOString(),
+              CONSTRUCTOR_ARGS:
+                `0x${data.result[0].ConstructorArguments}` || '0x',
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/utils/fetch-missing-deployments.ts:84]: Updated in another PR

File: script/utils/fetch-missing-deployments.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    for (const [contractName, contractAddress] of Object.entries(deployments)) {
+      try {
+        // Call explorer API
+        const url = new URL(networkConfig.explorerApiUrl)
+        url.searchParams.append('module', 'contract')
+        url.searchParams.append('action', 'getsourcecode')
+        url.searchParams.append('address', contractAddress as string)
+        url.searchParams.append('apiKey', apiKey)
+
+        const response = await fetch(url.toString())
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/utils/fetch-missing-deployments.ts:46]: Not needed

File: script/utils/fetch-missing-deployments.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        const data = await response.json()
+
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/utils/fetch-missing-deployments.ts:48]: It's fine

File: script/utils/fetch-missing-deployments.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        if (!data.result[0].SourceCode) {
+          console.log(`Skipping ${contractName}: No source code found`)
+          continue
+        }
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/utils/fetch-missing-deployments.ts:52]: If it fails it fails

**Pull request merged**